corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js

```diff
@@ -0,0 +1,97 @@
+        self.items = ko.observableArray();
+
+        self.init = function (config) {
+            self.padded_ns = {};
+            var max_ns_len = Math.max.apply(Math, _.map(config.namespaces, function (ns) { return ns.length }));
+            _(config.namespaces).each(function (namespace) {
+                var diff = max_ns_len - namespace.length,
+                    pad = new Array(diff + 1).join(PAD_CHAR);
+                self.padded_ns[namespace] = namespace + pad;
+            });
+            self.init_items(config);
+            self.latest_use = ko.observable(config.last_used._latest || '');
+        };
+
+        self.init_items = function (config) {
+            var items = config.items,
+                last_used = config.last_used || {};
+            self.items.removeAll();
+            _(items).each(function (item) {
+                var fields = item.split(':'),
+                    namespace = fields.length > 1 ? fields[0] : 'user',
+                    value = fields.length > 1 ? fields[1] : fields[0];
+                self.items.push({
+                    namespace: ko.observable(self.padded_ns[namespace]),
+                    value: ko.observable(value),
+                    last_used: ko.observable(last_used[value])
+                });
+            });
+        };
+
+        self.addItem = function (namespace) {
+            self.items.push({
+                namespace: ko.observable(self.padded_ns[namespace]),
+                value: ko.observable(),
+                last_used: ko.observable()
+            });
+            self.change();
+        };
+
+        self.removeItem = function (item) {
+            self.items.remove(item);
+            self.change();
+        };
+
+        self.change = function () {
+            self.saveButton.fire('change');
+        };
+
+        self.saveButton = COMMCAREHQ.SaveButton.init({
+            unsavedMessage: "You have unsaved changes",
+            save: function () {
+                var items = _.map(self.items(), function (item) {
+                    var ns_raw = item.namespace().replace(new RegExp(PAD_CHAR, 'g'), ''),
+                        namespace = ns_raw === 'user' ? null : ns_raw,
+                        value = namespace === null ? item.value() : namespace + ':' + item.value();
+                    return value;
+                });
+                self.saveButton.ajax({
+                    type: 'post',
+                    url: hqImport('hqwebapp/js/urllib.js').reverse('edit_toggle') + location.search,
+                    data: {
+                        item_list: JSON.stringify(items)
+                    },
+                    dataType: 'json',
+                    success: function (data) {
+                        self.init_items(data);
+                    }
+                });
+            }
+        });
+
+        var projectInfoUrl = '<a href="' + hqImport('hqwebapp/js/urllib.js').reverse('domain_internal_settings') + '">domain</a>';
```

Comment:
Agree this isn't great to read, but ultimately the template uses an `html` bind, so it does need a string, not an element. To be fair, that code could be reworked, but I'm trying not to make a whole lot of changes when splitting out js files (same reason that I'm ignoring the linter warnings).